### PR TITLE
ci: build and deploy with GitHub Actions

### PR DESCRIPTION
**Pull request by lwille**
related to INFRA-1257

### CODE REVIEW

**Comment by lwille** [comment]
hi @meyerdan (tagging you as main contributor): As you can see, we're moving from building the community projects on GitHub actions instead of "our" Jenkins.
We made this move to improve transparency as well as provide better feedback for contributors.

I'm going to leave this note in all affected repositories once we're done making the changes.
If you experience issues related to our changes to GitHub Actions, please let us know!

File: pom.xml [hunk 1]
```diff
@@ -8,7 +8,7 @@
   <parent>
     <groupId>org.camunda</groupId>
     <artifactId>camunda-release-parent</artifactId>
-    <version>2.5</version>
+    <version>3.7</version>
```
Comment (aabouzaid) [inline pom.xml:11]: Where this version comes from? :thinking:

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,103 @@
+# If this workflow is triggered by a push to master, it
```
Comment (aabouzaid) [inline .github/workflows/deploy.yml:1]: I don't have much experince with Github actions, but it looks good to me.

File: pom.xml (hunk 1 above)
Comment (lwille) [reply pom.xml:11]: it's the latest version of the `camunda-release-parent`. The commit message says:
`update camunda-release-parent@3.7 - required for deploying to both the Camunda Nexus and Maven Central`

The old version just doesn't support that.

**Review by aabouzaid — APPROVED** [review]

**Pull request merged**